dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/internal/GitRepositoryDetector.kt

```diff
@@ -3,14 +3,20 @@ package com.datadog.gradle.plugin.internal
 import com.datadog.gradle.plugin.DdAndroidGradlePlugin.Companion.LOGGER
 import com.datadog.gradle.plugin.RepositoryDetector
 import com.datadog.gradle.plugin.RepositoryInfo
+import com.datadog.gradle.plugin.internal.sanitizer.DdUrlSanitizer
+import com.datadog.gradle.plugin.internal.sanitizer.UrlSanitizer
 import java.io.File
 import org.gradle.api.Project
 import org.gradle.process.internal.ExecException
 
 // TODO RUMM-1095 handle git submodules
 // TODO RUMM-1096 handle git subtrees
 // TODO RUMM-1093 let customer override `origin` with custom remote name
-internal class GitRepositoryDetector : RepositoryDetector {
+internal class GitRepositoryDetector(
+    private val urlSanitizerFactory: () -> UrlSanitizer = {
```

Comment:
Why are you providing a factory and not an instance directly ?

Reply:
Good point...I needed this in a previous implementation but yes, not needed anymore